lib/adapters/mongodb.js

```diff
@@ -836,7 +837,7 @@ adapter._updateOneToMany = function(model, relatedModel, resource, reference, fi
       match[relatedModel.pk || "_id"] = resource[reference.path];
       if (resource._tenantId) match._tenantId = resource._tenantId;
 
-      return wrapAsyncCall(relatedModel, relatedModel.findOneAndUpdate, match, associate);
+      return wrapAsyncCall(relatedModel, relatedModel.findOneAndUpdate, match, associate, { new: true });
```

Comment:
Latest version of mongoose follows mongo native driver practice to return nothing after update (or return the data you just provided in the update). The param  `{ new: true }` forces it to re-read whatever data in database and return the record, as most of the fortune expects such a behaviour.